third_party/gif_encoder/src/main/java/com/bumptech/glide/gifencoder/AnimatedGifEncoder.java

```diff
@@ -252,15 +278,30 @@ public void setQuality(int quality) {
      *          int frame width.
      */
     public void setSize(int w, int h) {
-        if (started && !firstFrame)
+        if (started)
             return;
+
+        fixedWidth = w;
+        fixedHeight = h;
+        if (fixedWidth < 1)
+            fixedWidth = 320;
+        if (fixedHeight < 1)
+            fixedHeight = 240;
+
+        sizeSet = true;
+    }
+
+    /**
+     * Sets current GIF frame size.
```

Comment:
setFrameSize() sets the size of image for each frame. The width and height used here is the actual size of the image. This is a private function for internal use.
setSize() is a public function which will be used by some user who wants to set the size of the overall frame of the target GIF.

setSize was originally used for the two purposes mentioned above and we could not use different sizes for each frames. So I split it.